docs/docs/specs/v0.2/composer.md

```diff
@@ -195,10 +213,115 @@ Other plugins possible:
+
+## Hero Scenarios
+
+Priority legend:
+- P0 - must have
+- P1 - painful cut
+- P2 - nice to have
+- P3 - cut / next iteration
+
+### 5. Users can collaborate on documents in a stack-editor paradigm
+
+This scenario expands the plain-text-editor to support multimedia and other arbitrary content types by envisioning the document as a vertical sequence (**stack**) of **stack items**. 
+
+Users are free to drag-arrange section order, insert new sections, and interact with the contents of each section.
+
+This scenario demonstrates the value of `<Surface />` elements by delegating the presentation of each section to them, thereby taking advantage of plugins which may supply new content types for sections and components to render them with.
+
+Pure component:
+
+- 5.1 Developers can use a "pure", responsive `<Stack />` component to render a sequence of `<Stack.Item />` items
+- 5.2 Stack items provide mobile-friendly affordances for being dragged and re-ordered
+- 5.3 Stack provides an affordance for appending a new stack item
+- 5.4 Stack provides affordances for inserting new stack items between other items or at the top
+
+GFM in ECHO:
+
+- 5.5 Developers can use the `<MarkdownStackEditor />` (MSE) component for editing GFM Markdown documents in ECHO
+  - 5.5.1 MSE supports PlainText sections (P0)
+  - 5.5.2 MSE supports Image sections (P0)
+  - 5.5.3 MSE supports TaskList sections (P2)
+  - 5.5.4 MSE supports Table sections (P3)
+
+n.b.: until the various section types are supported, they appear as plain text.
+
+### 6. Users can collaborate on plain-text-files on disk
+
+This scenario demonstrates the power of bringing multiple data sources into the same application and enabling interactions between them.
+
+By bringing the local file system into Composer, we can shorten the distance between local files and ECHO spaces. Local files are the basis for successful app ecosystems from IDEs to modern productivity tools. ECHO spaces can make it easy to collaborate on local files using Composer.
+
+The key interactions to explore first are drag and drop operations between nodes from different data sources.
+
+Local files:
+
+- users can add a local disk folder to the Composer sidebar next to all the ECHO spaces
+- users can see the contents of those folders in Composer, and examine files by selecting them
+- details about unknown files are presented with a generic metadata presenter (file name, icon? only)
+- users can edit local known plain text files with the PlainText editor
+- users can see syntax highlighting for the file's language if supported (P2)
+- users can edit local markdown files with `MarkdownStackEditor`
+- local disk files require saving to persist
+- display of local folders chosen by users survives composer restart
+  - missing files do not show in the UI on restart (P0)
+  - missing files are shown in a disabled state which can be re-attached to another folder on disk (P3)
```

Comment:
what are "missing files" in this context?

Reply:
I thought it's possible that files on disk got deleted or moved while the app was down, and there would be cases where you'd wake up and fail to load previously granted folders